server/controllers/users-controller.js

```diff
@@ -0,0 +1,255 @@
+require('dotenv').config({ silent: true });
+const bCrypt = require('bcrypt'),
+  jwt = require('jsonwebtoken');
+
+const db = require('../models');
+/**
+ * UsersController class to handle all Users
+ * related actions
+ */
+class UsersController {
+
+  /**
+   * method fetchUser to create a User
+   * @param {object} userInfo - information about user
+   * @return {object} formatted object containing user information;
+   */
+  static fetchUser(userInfo) {
+    const { firstName, lastName, username, createdAt, email, id } = userInfo;
+    return { firstName, lastName, username, createdAt, email, id };
+  }
+
+  /**
+   * method createUser to create a User
+   * @param {object} req - request details
+   * @param {object} res - response details
+   * @return {object} response to the front end;
+   */
+  static createUser(req, res) {
+    db.user.create(req.body)
+      .then((userInfo) => {
+        const userDetails = UsersController.fetchUser(userInfo.dataValues);
+        UsersController.getUserRole(userDetails.id, (role) => {
+          const token = jwt.sign({ id: userDetails.id, role },
+            process.env.SECRET, { expiresIn: '24h' });
+          res.status(200).send({
+            done: true,
+            user: userDetails,
+            token
+          });
+        });
+      }).catch((error) => {
+        if (error.name === 'SequelizeForeignKeyConstraintError') {
+          return res.status(400).send({
+            done: false,
+            message: 'Role does not exist'
+          });
+        }
+        if (error.errors[0].type === 'unique violation') {
+          return res.status(400).send({ done: false,
+            message: `${error.errors[0].path} already exists` });
+        }
+        if (error.errors[0].type === 'notNull Violation') {
+          return res.status(400).send({ done: false,
+            message: `${error.errors[0].path} is required` });
+        }
+        res.status(500).send({ done: false });
+      });
+  }
+
+  /**
+   * method login to log a user in
+   * @param {object} req - request details
+   * @param {object} res - response details
+   * @return {object} jwtoken and status field;
+   */
+  static login(req, res) {
+    db.user.findAll({
+      where: {
+        username: req.body.username
+      },
+      include: [db.role]
+    }).then((userDetails) => {
+      if (userDetails[0]) {
+        const userInfo = UsersController.fetchUser(userDetails[0].dataValues);
```

Comment:
No, userInfo is just like a formatted userDetails. The `fetchUser` functions saves me the extra stress of writing userDetails[0].dataValues to get each property of the object